### PR TITLE
Add instructions for Amazon Linux 2023

### PR DESCRIPTION
**Pull request by nginx-seanmoloney**
### Proposed changes

Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue using one of the [supported keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) here in this description (not in the title of the PR).

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: site/content/installation-upgrade/installation-oss.md [hunk 1]
```diff
@@ -292,6 +292,38 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
+    module_hotfixes=true
+    ```
+
+1. To install `nginx-agent`, run the following command:
+
+    ```shell
+    sudo yum install nginx-agent
+    ```
+
+1. When prompted to accept the GPG key, verify that the fingerprint matches `573B FD6B 3D8F BC64 1079 A6AB ABF5 BD82 7BD9 BF62`, and if so, accept it.
```
Comment (y82) [inline site/content/installation-upgrade/installation-oss.md:320]:
```suggestion
1. When prompted to accept the GPG key, verify that the fingerprint matches
    `8540 A6F1 8833 A80E 9C16 53A4 2FD2 1310 B49F 6B46`,
    `573B FD6B 3D8F BC64 1079 A6AB ABF5 BD82 7BD9 BF62`,
    `9E9B E90E ACBC DE69 FE9B 204C BCDC D8A3 8D88 A2B3`
     and if so, accept it.
```

**Review by nginx-seanmoloney — COMMENTED** [review]
Looks good.

File: site/content/installation-upgrade/installation-oss.md [hunk 2]
```diff
@@ -292,6 +292,42 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
 
+### Installing NGINX Agent on Amazon Linux 2023
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo yum install yum-utils procps
+    ```
+
+1. To set up the yum repository for Amazon Linux 2, create the file named `/etc/yum.repos.d/nginx-agent.repo` with the following contents:
```
Comment (y82) [inline site/content/installation-upgrade/installation-oss.md:303]:
```suggestion
1. To set up the dnf repository for Amazon Linux 2023, create the file `/etc/yum.repos.d/nginx-agent.repo` with the following contents:
```

File: site/content/installation-upgrade/installation-plus.md [hunk 3]
```diff
@@ -380,7 +380,55 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
+1. Copy the `nginx-repo.crt` and `nginx-repo.key` files to the `/etc/ssl/nginx/` directory:
+
+    ```shell
+    sudo cp nginx-repo.crt nginx-repo.key /etc/ssl/nginx/
+    ```
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo yum install yum-utils procps ca-certificates
```
Comment (y82) [inline site/content/installation-upgrade/installation-plus.md:423]: sudo dnf install procps-ng ca-certificates (not sure about yum-utils and the exact name of procps)

File: site/content/installation-upgrade/installation-oss.md [hunk 4]
```diff
@@ -292,6 +292,42 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
 
+### Installing NGINX Agent on Amazon Linux 2023
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo dnf install yum-utils procps
+    ```
+
+1. To set up the yum repository for Amazon Linux 2, create the file named `/etc/yum.repos.d/nginx-agent.repo` with the following contents:
```
Comment (y82) [inline site/content/installation-upgrade/installation-oss.md:303]: Amazon Linux 2023

File: site/content/installation-upgrade/installation-oss.md [hunk 5]
```diff
@@ -292,6 +292,42 @@ Before you install NGINX Agent for the first time on your system, you need to se
     sudo nginx-agent -v
     ```
 
+### Installing NGINX Agent on Amazon Linux 2023
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo dnf install yum-utils procps
```
Comment (y82) [inline site/content/installation-upgrade/installation-oss.md:300]: not sure if yum-utils required with dnf. also procps name can be procps-ng - needs checking

File: site/content/installation-upgrade/installation-plus.md [hunk 6]
```diff
@@ -380,7 +380,55 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
+
+1. To set up the yum repository for Amazon Linux 2, create the file named `/etc/yum.repos.d/nginx-agent.repo` with the following contents:
+
+    ```
+    [nginx-agent]
+    name=nginx-agent repo
+    baseurl=https://pkgs.nginx.com/nginx-agent/amzn/2023/$releasever/$basearch
     sslclientcert=/etc/ssl/nginx/nginx-repo.crt
     sslclientkey=/etc/ssl/nginx/nginx-repo.key
     gpgcheck=0
```
[the remark this answers is not in the document]
Comment (nginx-seanmoloney) [reply site/content/installation-upgrade/installation-plus.md:435]: Using crt and key so gpgcheck can be 0

File: site/content/installation-upgrade/installation-plus.md [hunk 7]
```diff
@@ -380,7 +380,55 @@ Before you install NGINX Agent for the first time on your system, you need to se
     ```
     [nginx-agent]
     name=nginx-agent repo
-    baseurl=https://pkgs.nginx.com/nginx-agent/amzn2/$releasever/$basearch
+    baseurl=https://packages.nginx.org/nginx-agent/amzn/2023/$basearch/
+    sslclientcert=/etc/ssl/nginx/nginx-repo.crt
+    sslclientkey=/etc/ssl/nginx/nginx-repo.key
+    gpgcheck=0
```
[the remark this answers is not in the document]
Comment (nginx-seanmoloney) [reply site/content/installation-upgrade/installation-plus.md:387]: Same as above

File: site/content/installation-upgrade/installation-plus.md [hunk 8]
```diff
@@ -353,7 +354,55 @@ Before you install NGINX Agent for the first time on your system, you need to se
...
+1. Copy the `nginx-repo.crt` and `nginx-repo.key` files to the `/etc/ssl/nginx/` directory:
+
+    ```shell
+    sudo cp nginx-repo.crt nginx-repo.key /etc/ssl/nginx/
+    ```
+
+1. Install the prerequisites:
+
+    ```shell
+    sudo dnf install yum-utils procps ca-certificates
```
Comment (y82) [inline site/content/installation-upgrade/installation-plus.md:376]: procps-ng

**Review by y82 — APPROVED** [review]

**Review by ADubhlaoich — APPROVED** [review]
LGTM!

**Review by Jcahilltorre — APPROVED** [review]

**Pull request merged**